src/current/v23.1/cpu-profiler.md

```diff
@@ -0,0 +1,71 @@
+
+### Enabling automatic CPU profile capture
+
+- Set `server.cpu_profile.enabled` to `true`
+- Set `server.cpu_profile.cpu_usage_combined_threshold` to a value between `0` and `100`.
+
+{{site.data.alerts.callout_info}}
+Only setting `server.cpu_profile.enabled` to `true` will not generate a CPU profile.
+{{site.data.alerts.end}}
+
+### Recommended values
+
+- Set `server.cpu_profile.cpu_usage_combined_threshold` to `80` for 80%.
+- Set `server.cpu_profile.duration` to a lower value, for example `1s` or `2s`. This reduces [overhead](#overhead).
+- Set `server.cpu_profile.interval` to a lower value, for example `1m40s` (1 minute 40 seconds).
+
+### High-water mark threshold 
+
+The CPU Profiler runs asynchronously in the background. After every second, the Profiler checks if the CPU usage exceeds the high-water mark threshold. If so, it captures a CPU profile. If a profile capture is already in progress, a second profile is not taken.
+
+The Profiler uses the configuration options to determine the high-water mark threshold. For example, with `duration` set to `10s` , `interval` set to `5m0s`, and `cpu_usage_combined_threshold` set to `80`:
+
+- At `time0` the CPU usage polled is 82 percent which exceeds the baseline threshold of `80`, so a `10s` profile is captured and the high-water mark threshold becomes `82`.
+- After the `10s` profile capture, the Profiler continues to check every second if the CPU usage now exceeds `82` percent.
+- At `time1` the CPU usage polled is 85 percent, another profile is taken for `10s` and the high-water mark threshold becomes `85`.
+- At `time2`, the `5m0s` interval after `time0`, the high-water mark threshold is reset to the baseline threshold of `80`.
+- The Profiler continues its every second polling and captures profiles when CPU usage exceeds the high-water mark threshold beginning at the `80` percent baseline.
+
+## Accessing CPU profiles
+
+- The CPU profiler saves the captured CPU profiles to disk in the node filesystem. The default path is `/cockroach-data/logs/pprof_dump`.
+- The only way to get the CPU profiles is to access the node filesystem.
+- Enabling the CPU profiler does **not** add CPU profiles to [debug zips](cockroach-debug-zip.html).
+
+## Overhead
+
+- Monitor the following metrics:
+  - [P99 latency](ui-sql-dashboard.html#service-latency-sql-99th-percentile)
+  - P50 latency by creating a [custom chart](ui-custom-chart-debug-page.html) for the `sql.exec.latency-p50` metric
+  - [CPU usage](ui-hardware-dashboard.html#cpu-percent) 
+- A less than 10% regression on these foreground latency metrics indicates acceptable overhead.
+- The overhead is a cost to pay to collect the CPU profile needed to troubleshoot. Overhead only occurs during profile capture, not when it is idle. If the cluster can tolerate a 1 second increase in latency to capture the CPU profile, enable the CPU profiler.
```

Comment:
In a previous comment, I suggested combining the previous bullet with this bullet. 

I do like keeping this section, but have additional suggestions:

> Overhead only occurs during profile capture, not when it is idle. If the cluster can tolerate a 1 second increase in latency to capture the CPU profile, enable the CPU profiler.

^ Let's make the language around "1 second" more generic, since that depends on the value of `server.cpu_profile.duration`

Perhaps along the lines of: "Overhead only occurs during profile capture, not when it is idle. If the cluster can tolerate a `server.cpu_profile.duration` (e.g. 1 second) increase in latency to capture the CPU profile, `server.cpu_profile.duration` **consider** enabling the automatic CPU profiler."

Reply:
Done